streams/src/main/java/org/apache/kafka/streams/CloseOptions.java

```diff
@@ -81,7 +81,11 @@ public static CloseOptions groupMembershipOperation(final GroupMembershipOperati
      * @return this {@code CloseOptions} instance.
      */
     public CloseOptions withTimeout(final Duration timeout) {
-        this.timeout = Optional.ofNullable(timeout);
+        if (timeout == null) {
+            this.timeout = Optional.of(Duration.ofMillis(Long.MAX_VALUE));
```

Comment:
It looks like `timeout` is never `empty`. However, I think keeping the `empty` state is useful because it shows the user did not explicitly set a `timeout`. Maybe we can use `orElse` to address the comments

```java
        var timeout = optionsInternal.timeout().orElse(Duration.ofMillis(Long.MAX_VALUE));
        final String msgPrefix = prepareMillisCheckFailMsgPrefix(timeout, "timeout");
        final long timeoutMs = validateMillisecondDuration(timeout, msgPrefix);
```